src/geom_handlers/parq_site_assessment.jl

```diff
@@ -0,0 +1,153 @@
+"""Geometry-based assessment methods."""
+
+include("geom_ops.jl")
+include("common_assessment.jl")
+
+"""
+    assess_reef_site(
+        rel_pix::DataFrame,
+        geom::GI.Wrappers.Polygon,
+        max_count::Float64;
+        degree_step::Float64=15.0,
+        start_rot::Float64=0.0,
+        n_per_side::Int64=2,
+        surr_threshold::Float64=0.33
+    )::Tuple{Float64,Int64,GI.Wrappers.Polygon,Int64}
+
+Assesses the rotations of a search box `geom` for their pixel score. Returns the highest score,
+rotation step, polygon and a quality control flag for each assessment. Rotation steps are returned
+so that the `start_rot` angle is 0, rotations anti-clockwise are negative and rotations clockwise are
+positive.
+"""
+function assess_reef_site(
+    rel_pix::DataFrame,
+    geom::GI.Wrappers.Polygon,
+    max_count::Float64;
+    degree_step::Float64=15.0,
+    start_rot::Float64=0.0,
+    n_per_side::Int64=2,
+    surr_threshold::Float64=0.33
+)::Tuple{Float64,Int64,GI.Wrappers.Polygon,Int64}
+    rotations = (start_rot-(degree_step*n_per_side)):degree_step:(start_rot+(degree_step*n_per_side))
+    n_rotations = length(rotations)
+    score = zeros(n_rotations)
+    best_poly = Vector(undef, n_rotations)
+    qc_flag = zeros(Int64, n_rotations)
+
+    for (j, r) in enumerate(rotations)
+        rot_geom = rotate_geom(geom, r)
+        score[j] = size(rel_pix[GO.intersects.([rot_geom], rel_pix.geometry), :], 1) / max_count
+        best_poly[j] = rot_geom
+
+        if score[j] < surr_threshold
+            qc_flag[j] = 1
+            break
+        end
+    end
+
+    return score[argmax(score)], argmax(score)-(n_per_side+1), best_poly[argmax(score)], maximum(qc_flag)
+end
+
+"""
+    identify_potential_sites_edges(
+        df::DataFrame,
+        indices_pixels::Raster,
+        indices::Vector{CartesianIndex{2}},
```

Comment:
This is a good first pass but I think warrants additional thinking.

I'm not a big fan of these argument names as what they are for is not immediately obvious. 
It may have been me originally but now's the time to fix it up.

- `df` : What does the dataframe hold?
- `indices_pixels` : not 100% clear what it means
- `indices` of what? Valid locations (i.e., locations where we have data across all criteria)?

Apply the same kind of thinking to the other arguments as well please.

It's explained a bit in the docstring, but more informative names are always appreciated.

Note with 9 required arguments and 3 optional, it's on the very edge of what I would consider acceptable for function complexity (if the number of function arguments start to approach 8 or more, it starts to feel iffy as it suggests there is a cleaner way of doing things). 

The function seems to be doing more than one thing in the assessment (e.g., creating the search polygon).

You could create the polygon first and pass it into this function. Then in here, we can simply move/rotate it around.
At a quick glance it would reduce the number of arguments here by 3 or 4?

There are probably other things we could do to simplify too.

Reply:
I've renamed these arguments and changed it to take a processed dataframe containing the lon and lat values instead of doing that within the function. I was trying to do this to simplify the arguments the function needs as you say, but I realised I still need the resolution of the original raster data as well.